practices/follow-functional-core-imperative-shell.md

```diff
@@ -3,117 +3,153 @@
-However, fixating excessively on absolute purity may inadvertently introduce impractical code.
+However, excessively fixating on absolute purity may inadvertently introduce impractical code.
 For instance, rigidly avoiding variable mutation can sometimes lead to the use of recursion.
 While recursion aligns with functional purity, its efficiency may diminish, particularly in languages where stack management for large inputs poses challenges.
 Functional programming constructs, while elegant, may not always be the most efficient choice, especially in performance-critical scenarios.
 
-### Values are the Boundary
+### Remembering Values are the Boundary
+
+<ed> My goal with the above edit, and some of the below subhead edits, is to ensure the subhead is written in a way that implies it's a pitfall/challenge/limitation. Ok? </ed> 
 
 Values serve as the boundary between the layers.
 The imperative shell should communicate with the functional core by passing value objects exclusively, avoiding objects or functions that could potentially induce side effects.
 This ensures that the functional core remains isolated from external state changes, promoting clarity, predictability, and maintainability in the codebase while facilitating easier testing, debugging, and refactoring.
 
-### Learning Curve
+### Potentially Steep Learning Curve
 
-Transitioning to the Functional Core, Imperative Shell practice may present a learning curve.
+Transitioning to the Functional Core, Imperative Shell practice may present a steep learning curve for teams.
 To facilitate this transition smoothly, it's recommended that developers with more knowledge mentor other developers through pair programming sessions.
 Additionally, fostering an environment of knowledge sharing, providing resources, and allocating time for developers to study the practice can greatly aid in its adoption and understanding across the team.
 
 ### Is Incompatible with Object-Oriented Programming
 
-The Functional Core, Imperative Shell practice is not mutually exclusive with Object-Oriented Programming (OOP) principles.
-As long as the functional core remains free of side effects, it can coexist within an object-oriented architecture.
-The Functional Core's logic could be encapsulated within methods of a class, ensuring that these methods return new objects instead of mutating their fields.
-The Functional Core can take advantage of polymorphism by receiving or returning objects of a class hierarchy with multiple subclasses to represent various inputs or outputs.
-Developers can implement the functional core while adhering to OOP principles and have the advantages of both paradigms.
+<ed> I'm having trouble understanding the pitfall/challenge/limitation with this one. The text below implies developers can take advantage of both paradigms. Should we rephrase this subhead? Seems like "Can Be Compatible with OOP" is more in line with the text below. </ed> 
```

Comment:
Yeah, "Is Incompatible with Object-Oriented Programming" is more of a popular and untrue belief. I'm fine with your suggested title.